assignments/array-methods.js

```diff
@@ -1,83 +1,556 @@
+
+runners.forEach(function({ first_name, last_name }) {
+  fullName.push(`${first_name} ${last_name}`);
+});
+
+console.log("fullName", fullName);
 
 // ==== Challenge 2: Use .map() ====
-// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
+// The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER.
+//  Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+const needPush = runners.map(function({ first_name }) {
+  return first_name.toUpperCase();
+});
+
+allCaps.push(needPush);
+
+console.log("allCaps", allCaps);
+
+// Previous Clean Example:
+
+// let carYears = inventory.map(car => {
+//     return car.car_year;
+//   });
+//   console.log(carYears);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
-console.log(largeShirts);
+// The large shirts won't be available for the event due to an ordering issue.
+// Get a list of runners with large sized shirts so they can choose a different size.
+// Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+
+const largeShirts = runners.filter(function(shirtSize) {
+  return shirtSize.shirt_size == "L";
+});
+
+console.log("largeShirts", largeShirts);
+
+// Previous Clean Example:
+
+// let oldCars = carYears.filter(function(carAge) {
+//     return carAge <= 2000;
+//   });
+
+//   console.log(oldCars.length);
 
 // ==== Challenge 4: Use .reduce() ====
-// The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+// The donations need to be tallied up and reported for tax purposes.
+// Add up all the donations into a ticketPriceTotal array and log the result
+let ticketPriceTotal = runners.reduce(function(acc, money) {
+  return (acc += money.donation);
+}, 0);
+
+console.log("ticketPriceTotal:", ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 - Display all companies
+
+whatCompany = [];
+const companyNames = runners.map(function(theCompanies) {
+  return theCompanies.company_name;
+});
+whatCompany.push(companyNames);
```

Comment:
Remember .map() returns a new array of the same size, therefore you could make this a bit cleaner by doing

```
whatCompany = runners.map(function(theCompanies) {
  return theCompanies.company_name;
});

console.log(whatCompany)
```

Typically you'd use `.push()` when you're using an array method like `.forEach()`, `.forEach()` and `.map()` are pretty much the exact same thing but the difference is `.map()` returns a new array and with `.forEach()` you would need to use `.push()` to push the current item into an existing array.